python/taichi/lang/impl.py

```diff
@@ -854,9 +864,24 @@ def fused_string(entries):
         if accumated:
             yield accumated
 
+    def extract_formats(entries):
+        contents = []
+        formats = []
+        for entry in entries:
+            if isinstance(entry, list):
```

Comment:
Let's make `entry2content` have consistent return types so the `entry` here doesn't have dynamic type.

Reply:
`entry2content()` now has consistent return type~